pkg/apis/extensions/validation/validation.go

```diff
@@ -291,6 +291,10 @@ func ValidateDeploymentSpec(spec *extensions.DeploymentSpec, fldPath *field.Path
 	allErrs = append(allErrs, apivalidation.ValidatePositiveField(int64(spec.Replicas), fldPath.Child("replicas"))...)
 	allErrs = append(allErrs, apivalidation.ValidatePodTemplateSpecForRC(&spec.Template, spec.Selector, spec.Replicas, fldPath.Child("template"))...)
 	allErrs = append(allErrs, ValidateDeploymentStrategy(&spec.Strategy, fldPath.Child("strategy"))...)
+	// nil pointer is a valid RevisionHistoryLimit
```

Comment:
nil pointer isn't a valid `RevisionHistoryLimit`, it just means unspecified. Zero is a valid value